modules/internal/DefaultAssociative.chpl

```diff
@@ -566,14 +566,15 @@ module DefaultAssociative {
 
     // ref version
     proc dsiAccess(idx : idxType, haveLock = false) ref {
-      const shouldLock = dom.parSafe && !haveLock;
+      const arrOwnsDom = dom._arrs.length == 1;
+      const shouldLock = dom.parSafe && !haveLock && arrOwnsDom;
```

Comment:
@mppf: On reflection, I'm not sure that this was a good change to make.  Imagine the following code:

```chapel
forall key in collection do
  A[key] += 1;
```

If we only lock writes to `A` which resize the table, then any writes to `A` which happen to be to an existing key may happen in an unlocked manner and therefore may run afoul of a simultaneous resizing being done by another write.  This would suggest that all write accesses to `A` should lock, whether or not they result in a resizing.  But we could restrict the locking to writes to `A` which have the 1:1 domain:array property (since writes to other associative cases won't resize but will result in an OOB error).  Do you buy that?

Ideally, we could also avoid the lock in write accesses of associative arrays with const domains (since they also can't be resized), but I don't think we can tell whether an array's domain is const or not from within module code.